### PR TITLE
Fixes #5522 by implementing timeout for the profiler

### LINKED ISSUE #5522: Profiler Timeout Logic is Not Working
**Affected module**
Profiler

**Describe the bug**
Current implementation of the timeout does not work properly when executing queries with SQL Alchemy

**To Reproduce**
Run a profiler query lasting more than 10 minutes

**Expected behavior**
By default the timeout is set at 10 minutes. The query should timeout and the session be closed once this timeout value is reached

**Additional context**
We need to investigate what is preventing SQL Alchemy from picking up the timeout logic and implement a fix

### PR DESCRIPTION
**Pull request by TeddyCr**
### Describe your changes :
Fixes #5522 

We added a timeout logic in the profiler workflow. Timeout defaults to 12 hours and is set in seconds. Due to python threading implementation if timeout is reached, the profiler will still have to wait for active threads to finished (either by failing or completing) before it can kill the whole threadpool and move on to execute against the next table.

#
### Type of change :

- [ ] Bug fix
- [x] Improvement
- [ ] New feature
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation

#
### Frontend Preview (Screenshots) :
Screenshot 2022-12-16 at 10 51 16 AM

### CODE REVIEW

**Review by chirag-madlani — DISMISSED** [review]
Looks good on UI side

File: openmetadata-ui/src/main/resources/ui/src/components/AddIngestion/Steps/ConfigureIngestion.tsx [hunk 1]
```diff
@@ -180,6 +182,29 @@ const ConfigureIngestion = ({
...
   };
 
+  const getTimeoutSeconds = () => {
+    return (
+      <div>
+        <label>Timeout in Seconds </label>
+        <p className="tw-text-grey-muted tw-mt-1 tw-mb-2 tw-text-sm">
+          Optional number setting the timeout in seconds for the profiler. If
+          the timeout is reached the profiler will wait for any pending queries
+          to terminated its execution.
```
Comment (Sachin-chaurasiya) [inline openmetadata-ui/src/main/resources/ui/src/components/AddIngestion/Steps/ConfigureIngestion.tsx:192]: Can we move this to locale files and read from there?

File: openmetadata-ui/src/main/resources/ui/src/components/AddIngestion/Steps/ConfigureIngestion.tsx [hunk 2]
```diff
@@ -180,6 +182,29 @@ const ConfigureIngestion = ({
     );
   };
 
+  const getTimeoutSeconds = () => {
+    return (
+      <div>
+        <label>Timeout in Seconds </label>
```
Comment (Sachin-chaurasiya) [inline openmetadata-ui/src/main/resources/ui/src/components/AddIngestion/Steps/ConfigureIngestion.tsx:188]: Can we move this to locale files and read from there?

**Review by Sachin-chaurasiya — APPROVED** [review]

File: ingestion/src/metadata/interfaces/sqalchemy/sqa_profiler_interface.py [hunk 3]
```diff
@@ -369,35 +372,45 @@ def get_all_metrics(
...
+                            {
+                                "name": column,
+                                "timestamp": datetime.now(tz=timezone.utc).timestamp(),
+                                **profile,
+                            }
+                        )
+                except concurrent.futures.TimeoutError as exc:
+                    pool.shutdown39(wait=True, cancel_futures=True)
+                    logger.debug(traceback.format_exc())
+                    logger.info(f"Operation was cancelled due to TimeoutError - {exc}")
```
Comment (nahuelverdugo) [inline ingestion/src/metadata/interfaces/sqalchemy/sqa_profiler_interface.py:411]: Not sure if an error is better here:
```suggestion
                    logger.error(f"Operation was cancelled due to TimeoutError - {exc}")
```

**Review by nahuelverdugo — APPROVED** [review]

**Review by nahuelverdugo — APPROVED** [review]

**Pull request merged**